src/openforms/logging/logevent.py

```diff
@@ -441,18 +449,94 @@ def submission_export_list(form: "Form", user: "User"):
 def submission_logic_evaluated(
     submission: "Submission", evaluated_rules, updated_submission_data
 ):
-
     if not evaluated_rules:
         return
     log_evaluated_rules = []
+    formsteps = submission.form.formstep_set.all()
+    # Mapping each form's key to :
+    # - the form_step it belongs
+    # - The component related to it
+    keys_mapped_to_form = {}
+    keys_mapped_to_component = {}
+    for formstep in formsteps:
+        keys_mapped_to_form = {
+            **{component["key"]: formstep for component in formstep.iter_components()},
+            **keys_mapped_to_form,
+        }
+        keys_mapped_to_component = {
+            **flatten(formstep.form_definition.configuration),
+            **keys_mapped_to_component,
+        }
 
+    involved_components = {}
     for evaluated_rule in evaluated_rules:
 
         rule = evaluated_rule["rule"]
+        logic_data = JsonLogicTest.from_expression(rule.json_logic_trigger)
+        source_components = convert_json_logic_in_json_tree(
+            logic_data,
+            keys_mapped_to_component,
+            keys_mapped_to_form,
+            updated_submission_data,
+        )
+
+        stringified_json_tree = stringify_json_tree(source_components)
+
+        # will remove duplicate components that are involved in several rules
+        remove_duplicate_components_from_json_tree(
+            source_components, involved_components
+        )
```

Comment:
I don't understand what this does - can you explain?

Reply:
ah okay, it's de-duplication for the "variables" section then if I understand it correctly?

Reply:
Yes, exactly